packages/eslint-plugin/tests/rules/no-shadow/no-shadow.test.ts

```diff
@@ -547,6 +547,364 @@ let y;
       languageOptions: { parserOptions: { ecmaVersion: 6 } },
       options: [{ hoist: 'all' }],
     },
+    {
+      code: `
+let x = foo((x, y) => {});
+let y;
+      `,
+      errors: [
+        {
+          data: {
+            name: 'x',
+            shadowedColumn: 5,
+            shadowedLine: 2,
+          },
+          messageId: 'noShadow',
+          type: AST_NODE_TYPES.Identifier,
+        },
+      ],
+      languageOptions: { parserOptions: { ecmaVersion: 6 } },
+      options: [{ hoist: 'functions' }],
+    },
```

Comment:
This isn't directly related to the PR, but it seemed there are very few tests that cover the `functions` option in `hoist` (which now has two extra options).

Reply:
👍👍👍 on adding in missing test coverage! Especially if it's at all related to a rule being changed.